updates/pg_partman--4.7.3--5.0.0-beta.sql

```diff
@@ -5462,6 +5452,54 @@ END;
 $$;
 
 
+CREATE PROCEDURE @extschema@.run_maintenance_proc(
+    p_wait int DEFAULT 0
+    -- Keep these defaults in sync with `run_maintenance`!
+    , p_analyze boolean DEFAULT false
+    , p_jobmon boolean DEFAULT true
+)
+    LANGUAGE plpgsql
+    AS $$
+DECLARE
+
+v_adv_lock              boolean;
+p_parent_table          text;
+
+BEGIN
+
+v_adv_lock := pg_try_advisory_lock(hashtext('pg_partman run_maintenance'));
+IF v_adv_lock = false THEN
+    RAISE NOTICE 'Partman maintenance already running or another session has not released its advisory lock.';
+    RETURN;
+END IF;
+
+FOR p_parent_table IN
+    SELECT parent_table
+    FROM @extschema@.part_config
+    WHERE undo_in_progress = false
+    AND automatic_maintenance = 'on'
+LOOP
+/*
+ * Run maintenance with a commit between each partition set
+ * TODO - Once PG11 is more mainstream, see about more full conversion of run_maintenance function as well as turning
+ *        create_partition* functions into procedures to commit after every child table is made. May need to wait
+ *        for more PROCEDURE features as well (return values, search_path, etc).
+ *      - Also see about swapping names so this is the main object to call for maintenance instead of a function.
+ */
+    RAISE DEBUG 'run_maintenance_proc for table: %', p_parent_table;
+    PERFORM @extschema@.run_maintenance(p_parent_table, p_jobmon => p_jobmon, p_analyze => p_analyze);
```

Comment:
So I kind of wanted the actual, full call of the function to be part of the DEBUG statement here. So I know all the parameters that were passed as part of the call. Makes it a bit more complicated to create the SQL but better for debugging.

Reply:
If the problem happens back in the actual caller of the function due to some bad value being passed in, then the call stack will never reach the called function to be debugged. And call tracing is particularly bad in plgsql because you'll get such lovely simplistic errors as `ERROR at .` with no further reference to line number. You do get a character count number in the log sometimes but even that can often be several lines off where the actual issue is.

I don't get the dislike of building SQL in order to debug. I know it should be done carefully with format. Also I know DEBUG is usually disabled, but the entire reason it's there is so that it can be enabled when it is needed. If debugging were better in plgsql I could see your point, but when building more complicated SQL statements, having that SQL output during debug parsing can be greatly useful and I often have DEBUG enabled during development for that purpose.This particular instance isn't that complicated, so just debugging the parameters should probably be ok. But I don't think the practice should be summarily dismissed.

Reply:
Will be reverting this debug change after merging